### ISSUE TITLE
Aggregations don't work if a type has the same name as a field in an other type

### ISSUE THREAD

**Issue by michaelperrin** [elastic/elasticsearch#16247]
Hi,

Here is a bug that is always reproducible on Elasticsearch 1.7.4 (no problem using 2.1.1). It concerns aggregations that return no results (while it should) when a type has the same name as a field name of an other type.

Here is a way to reproduce systematically the issue:

```
# If you just change "state" with "state2" in the second type, everything will work fine
curl -XPUT 'http://localhost:9200/agg_analysis' -d '
{
  "mappings": {
    "data": {
      "properties": {
        "state" : {
          "type": "string",
          "fields": {
            "raw" : {
              "type": "string",
              "index": "not_analyzed"
            }
          }
        }
      }
    },
    "state": {
      "properties": {
        "name" : {
          "type": "string",
          "fields": {
            "raw" : {
              "type": "string",
              "index": "not_analyzed"
            }
          }
        }
      }
    }
  }
}
'
```

```
curl -XPOST 'http://localhost:9200/agg_analysis/data/_bulk' -d '
{ "index": {}}
{ "state" : "New York" }
{ "index": {}}
{ "state" : "New Mexico" }
{ "index": {}}
{ "state" : "New York" }
'
```

```
curl -XGET 'http://localhost:9200/agg_analysis/data/_search?pretty' -d '
{
  "size" : 0,
  "aggs" : {
    "states" : {
        "terms" : {
            "field" : "state.raw"
        }
    }
  }
}
'
```

This should return a list of 2 states, but returns none. Note that `state` is:

* The name of a field in the `data` type.
* And also the name of an other type.

Changing the type name from `state` to `state2` in the mapping will make the aggregation on the `data` type work which is really weird.

To check this out, you can run all the steps again by replacing from `state` with `state2` in the mapping step.

**Comment by rjernst** [comment]
This was fixed with 2.0, in #8872, and was part of a very large set of breaking changes to the mappings api.

**Comment by michaelperrin** [comment]
Thanks @rjernst for your answer!

As I am stuck with Elasticsearch 1.7.4 for now (as the plugin (aka. bundle) for Symfony2 is not entirely ready yet for version 2), I will use the full path with the type prepended in my aggregatio (in my example: `data.state.raw`), even if it is incompatible with ES 2.x . I will change it back to `data.state` when I upgrade to ES 2.
